core/src/repositories/cc/Repository.cc

```diff
@@ -6,15 +6,24 @@
 #include <string>
 
 Repository::Repository(std::string socket_path)
```

Comment:
Taking a look through this class. I think many of the message sending functions should be relegated to a single interface instead of being combined with data storage. Similar to the above, this appears to violate the [Single Responsibility principal](https://en.wikipedia.org/wiki/Single_responsibility_principle)

Reply:
Hmm I may have to get Repository to use a UnixDomainSocketClient rather than inherit from it.

Reply:
We could keep repository as the class where messages come in and get processed, then have another class that handles the replys? That would keep the purpose of repository as processing each message, And if a reply is needed we ship it off to another class? Thoughts?

Reply:
I'm going to revisit this through an architectural perspective